packages/application-generic/src/services/cache/key-builders/entities.ts

```diff
@@ -23,6 +23,21 @@ const buildSubscriberKey = ({
     identifier: subscriberId,
   });
 
+const buildSubscriberIdentifierKey = ({
+  _id,
+  _environmentId,
+}: {
+  _id: string;
+  _environmentId: string;
+}): string =>
+  buildCommonKey({
+    type: CacheKeyTypeEnum.ENTITY,
+    keyEntity: CacheKeyPrefixEnum.SUBSCRIBER,
+    environmentId: _environmentId,
+    identifierPrefix: IdentifierPrefixEnum.ID,
+    identifier: _id,
+  });
+
```

Comment:
I understand why you took this approach :) But adding another caching function is not preferred for this particular issue. Handling cache is delicate, we would need to add invalidation at places, and also it will not use the already cached data we have by the subscriberId. So, as it is done only for the actor (the only place its cached by _id) it will probably not have significant affect. Yet, using the one with subscriberId will have so much affect, as we know the actor has already been fetched once in trigger-event.
Let's go the other way, and add the external subscriberId to the job entity. I think we should call it actorSubscriberId. 
it will be the equivalent of :
 _id -> _actorId and subscriberId -> actorSubscriberId
 Let me know if that makes sense to you